nuxeo-services/nuxeo-platform-convert/src/main/java/org/nuxeo/ecm/platform/convert/plugins/UTF8CharsetConverter.java

```diff
@@ -95,7 +95,7 @@ protected Blob convert(Blob blob) throws IOException, ConversionException {
         return newBlob;
     }
 
-    protected String detectEncoding(InputStream in) throws IOException, ConversionException {
+    protected String detectEncoding(Blob blob, InputStream in) throws IOException, ConversionException {
```

Comment:
I would also say to deprecate and add a new method, even it's only protected and not public, it's a rule we generally follow to not potentially break sub-classes.

Reply:
I asked myself if i should change the method and the call.

@ataillefer : Yes to keep the two methods and deprecate the original one, but we will end with duplicate code warning. 
does it not matter ?

Reply:
Not if one method calls the other one no?

Reply:
@ataillefer : It better to deprecate this method,  because the param it a type of InputStream and it's protected. it a bad idea from my point of view. Because it not close the InputStream. The client must close it and we are not sur if it will do it. But i think it will not clear if i deprecated a method and just after that i use it :). I prefer to create a new method  which takes a InputStream (with private visibility) and both methods (new and the original) call this one. I push this code et you will tell me what do you think

Reply:
You shouldn't use private visibility in Nuxeo, we use protected instead.

Reply:
yes, usually i use the protected one. but i was wondering about the case where calling this method but without closing the InputStream. that enforce the using for the one with Blob params and it can be override.